### PR TITLE
fix: cgo support

### PR DESCRIPTION
**Pull request by ldez**
This PR handle cgo without hack.

### CODE REVIEW

File: getters.go [hunk 1]
```diff
@@ -275,11 +262,11 @@ func matchAny(s string, rr []*regexp.Regexp) bool {
...
 }
 
-func isCgoGenerated(lines []string) bool {
-	for i := range lines {
-		if strings.Contains(lines[i], "Code generated by cmd/cgo") {
-			return true
-		}
+func getFilename(fset *token.FileSet, file *ast.File) (string, bool) {
+	filename := fset.PositionFor(file.Pos(), true).Filename
+	if !strings.HasSuffix(filename, ".go") {
```
Comment (tetafro) [inline getters.go:267]: Could you explain what's going on here? What kind of files don't have ".go" suffix?

File: getters.go [hunk 2]
```diff
@@ -36,36 +36,23 @@ func newParsedFile(file *ast.File, fset *token.FileSet) (*parsedFile, error) {
...
-	// new generated comments at the top. Because of them the content differs
-	// from AST. For some reason it differs only in golangci-lint. I failed to
-	// find out the exact description of the process, so let's just skip files
-	// generated by cgo.
-	if isCgoGenerated(pf.lines) {
-		return nil, errUnsuitableInput
+	var filename string
+	filename, _ = getFilename(fset, file)
+
+	if !strings.HasSuffix(filename, ".go") {
```
Comment (tetafro) [inline getters.go:47]: That's already checked in `getFilename`, right?

Comment (ldez) [reply getters.go:47]: No, in `getFilename` I check only the filename at the adjusted position.

Here I evaluate the filename at the adjusted position and the non-adjusted position.

File: getters.go (hunk 1 above)
Comment (ldez) [reply getters.go:267]: A lot of files don't have `.go` extension:
- compilation artifacts (often related to cgo)
- file used by a line directive (`//line hello.tmpl:1`)
- etc.

<details>

```
adjusted position: /home/ldez/sources/golangci/sandbox/line/hello.tmpl non adjusted position: /home/ldez/sources/golangci/sandbox/line/line.go

adjusted position: /home/ldez/sources/golangci/sandbox/main.go non adjusted position: /home/ldez/sources/golangci/sandbox/main.go

adjusted position file: /home/ldez/sources/golangci/sandbox/line/hello.tmpl 
 # This is a template file for some code generator, and is not a valid go source file.

non adjusted position file: /home/ldez/sources/golangci/sandbox/line/line.go 
 //line hello.tmpl:1
package line

func Foo() {

}

---------
adjusted position: /home/ldez/.cache/go-build/62/6248342cd6ede467fbde8a115d730d771a2fcb65488a2bf6ac6dd981cfcf0e6b-d non adjusted position: /home/ldez/.cache/go-build/62/6248342cd6ede467fbde8a115d730d771a2fcb65488a2bf6ac6dd981cfcf0e6b-d

adjusted position file: /home/ldez/sources/golangci/sandbox/main.go 
 package main

import (
        "os"

        "golang.org/x/tools/go/analysis"
        "golang.org/x/tools/go/analysis/singlechecker"
)

var Analyzer = &analysis.Analyzer{
        Name: "example",
        Doc:  "example",
        Run:  run,
}

func run(pass *analysis.Pass) (interface{}, error) {
        for _, file := range pass.Files {
                adj := pass.Fset.PositionFor(file.Pos(), true).Filename
                nonAdj := pass.Fset.PositionFor(file.Pos(), false).Filename

                println("adjusted position:", adj, "non adjusted position:", nonAdj)
                println()

                data, err := os.ReadFile(adj)
                if err != nil {
                        return nil, err
                }

                // println("adjusted position:", adj)
                println("adjusted position file:", adj, "\n", string(data))

                data, err = os.ReadFile(nonAdj)
                if err != nil {
                        return nil, err
                }

                // println("non adjusted position:", nonAdj)
                println("non adjusted position file:", nonAdj, "\n", string(data))

                println("---------")
        }

        return nil, nil
}

func main() {
        singlechecker.Main(Analyzer)
}

adjusted position file: /home/ldez/.cache/go-build/62/6248342cd6ede467fbde8a115d730d771a2fcb65488a2bf6ac6dd981cfcf0e6b-d 
 //go:cgo_ldflag "-O2"
//go:cgo_ldflag "-g"
// Code generated by cmd/cgo; DO NOT EDIT.

package cgo

import "unsafe"

import "syscall"

import _cgopackage "runtime/cgo"

type _ _cgopackage.Incomplete
var _ syscall.Errno
func _Cgo_ptr(ptr unsafe.Pointer) unsafe.Pointer { return ptr }

//go:linkname _Cgo_always_false runtime.cgoAlwaysFalse
var _Cgo_always_false bool
//go:linkname _Cgo_use runtime.cgoUse
func _Cgo_use(interface{})
//go:linkname _Cgo_no_callback runtime.cgoNoCallback
func _Cgo_no_callback(bool)
type _Ctype_double float64

type _Ctype_void [0]byte

//go:linkname _cgo_runtime_cgocall runtime.cgocall
func _cgo_runtime_cgocall(unsafe.Pointer, uintptr) int32

//go:linkname _cgoCheckPointer runtime.cgoCheckPointer
//go:noescape
func _cgoCheckPointer(interface{}, interface{})

//go:linkname _cgoCheckResult runtime.cgoCheckResult
//go:noescape
func _cgoCheckResult(interface{})

//go:cgo_import_static _cgo_70b3d9cc8937_Cfunc_add
//go:linkname __cgofn__cgo_70b3d9cc8937_Cfunc_add _cgo_70b3d9cc8937_Cfunc_add
var __cgofn__cgo_70b3d9cc8937_Cfunc_add byte
var _cgo_70b3d9cc8937_Cfunc_add = unsafe.Pointer(&__cgofn__cgo_70b3d9cc8937_Cfunc_add)

//go:cgo_unsafe_args
func _Cfunc_add(p0 _Ctype_double, p1 _Ctype_double) (r1 _Ctype_double) {
        _cgo_runtime_cgocall(_cgo_70b3d9cc8937_Cfunc_add, uintptr(unsafe.Pointer(&p0)))
        if _Cgo_always_false {
                _Cgo_use(p0)
                _Cgo_use(p1)
        }
        return
}
//go:cgo_import_static _cgo_70b3d9cc8937_Cfunc_floor
//go:linkname __cgofn__cgo_70b3d9cc8937_Cfunc_floor _cgo_70b3d9cc8937_Cfunc_floor
var __cgofn__cgo_70b3d9cc8937_Cfunc_floor byte
var _cgo_70b3d9cc8937_Cfunc_floor = unsafe.Pointer(&__cgofn__cgo_70b3d9cc8937_Cfunc_floor)

//go:cgo_unsafe_args
func _Cfunc_floor(p0 _Ctype_double) (r1 _Ctype_double) {
        _cgo_runtime_cgocall(_cgo_70b3d9cc8937_Cfunc_floor, uintptr(unsafe.Pointer(&p0)))
        if _Cgo_always_false {
                _Cgo_use(p0)
        }
        return
}

non adjusted position file: /home/ldez/sources/golangci/sandbox/main.go 
 package main

import (
        "os"

        "golang.org/x/tools/go/analysis"
        "golang.org/x/tools/go/analysis/singlechecker"
)

var Analyzer = &analysis.Analyzer{
        Name: "example",
        Doc:  "example",
        Run:  run,
}

func run(pass *analysis.Pass) (interface{}, error) {
        for _, file := range pass.Files {
                adj := pass.Fset.PositionFor(file.Pos(), true).Filename
                nonAdj := pass.Fset.PositionFor(file.Pos(), false).Filename

                println("adjusted position:", adj, "non adjusted position:", nonAdj)
                println()

                data, err := os.ReadFile(adj)
                if err != nil {
                        return nil, err
                }

                // println("adjusted position:", adj)
                println("adjusted position file:", adj, "\n", string(data))

                data, err = os.ReadFile(nonAdj)
                if err != nil {
                        return nil, err
                }

                // println("non adjusted position:", nonAdj)
                println("non adjusted position file:", nonAdj, "\n", string(data))

                println("---------")
        }

        return nil, nil
}

func main() {
        singlechecker.Main(Analyzer)
}

---------
non adjusted position file: /home/ldez/.cache/go-build/62/6248342cd6ede467fbde8a115d730d771a2fcb65488a2bf6ac6dd981cfcf0e6b-d 
 //go:cgo_ldflag "-O2"
//go:cgo_ldflag "-g"
// Code generated by cmd/cgo; DO NOT EDIT.

package cgo

import "unsafe"

import "syscall"

import _cgopackage "runtime/cgo"

type _ _cgopackage.Incomplete
var _ syscall.Errno
func _Cgo_ptr(ptr unsafe.Pointer) unsafe.Pointer { return ptr }

//go:linkname _Cgo_always_false runtime.cgoAlwaysFalse
var _Cgo_always_false bool
//go:linkname _Cgo_use runtime.cgoUse
func _Cgo_use(interface{})
//go:linkname _Cgo_no_callback runtime.cgoNoCallback
func _Cgo_no_callback(bool)
type _Ctype_double float64

type _Ctype_void [0]byte

//go:linkname _cgo_runtime_cgocall runtime.cgocall
func _cgo_runtime_cgocall(unsafe.Pointer, uintptr) int32

//go:linkname _cgoCheckPointer runtime.cgoCheckPointer
//go:noescape
func _cgoCheckPointer(interface{}, interface{})

//go:linkname _cgoCheckResult runtime.cgoCheckResult
//go:noescape
func _cgoCheckResult(interface{})

//go:cgo_import_static _cgo_70b3d9cc8937_Cfunc_add
//go:linkname __cgofn__cgo_70b3d9cc8937_Cfunc_add _cgo_70b3d9cc8937_Cfunc_add
var __cgofn__cgo_70b3d9cc8937_Cfunc_add byte
var _cgo_70b3d9cc8937_Cfunc_add = unsafe.Pointer(&__cgofn__cgo_70b3d9cc8937_Cfunc_add)

//go:cgo_unsafe_args
func _Cfunc_add(p0 _Ctype_double, p1 _Ctype_double) (r1 _Ctype_double) {
        _cgo_runtime_cgocall(_cgo_70b3d9cc8937_Cfunc_add, uintptr(unsafe.Pointer(&p0)))
        if _Cgo_always_false {
                _Cgo_use(p0)
                _Cgo_use(p1)
        }
        return
}
//go:cgo_import_static _cgo_70b3d9cc8937_Cfunc_floor
//go:linkname __cgofn__cgo_70b3d9cc8937_Cfunc_floor _cgo_70b3d9cc8937_Cfunc_floor
var __cgofn__cgo_70b3d9cc8937_Cfunc_floor byte
var _cgo_70b3d9cc8937_Cfunc_floor = unsafe.Pointer(&__cgofn__cgo_70b3d9cc8937_Cfunc_floor)

//go:cgo_unsafe_args
func _Cfunc_floor(p0 _Ctype_double) (r1 _Ctype_double) {
        _cgo_runtime_cgocall(_cgo_70b3d9cc8937_Cfunc_floor, uintptr(unsafe.Pointer(&p0)))
        if _Cgo_always_false {
                _Cgo_use(p0)
        }
        return
}

---------
adjusted position: /home/ldez/sources/golangci/sandbox/cgo/cgo.go non adjusted position: /home/ldez/.cache/go-build/d2/d202ddb4ff2eb75186fb85b2d45b3abbede18e343180e149a0ae28f0849fa877-d

adjusted position file: /home/ldez/sources/golangci/sandbox/cgo/cgo.go 
 package cgo

/*
#include "math.h"
double add(double a, double b) {
    return a + b;
}
*/
import "C"
import "fmt"

func Foo() {
        fmt.Println(C.floor(C.add(1, 2.1)))
}

non adjusted position file: /home/ldez/.cache/go-build/d2/d202ddb4ff2eb75186fb85b2d45b3abbede18e343180e149a0ae28f0849fa877-d 
 // Code generated by cmd/cgo; DO NOT EDIT.

//line /home/ldez/sources/golangci/sandbox/cgo/cgo.go:1:1
package cgo

/*
#include "math.h"
double add(double a, double b) {
    return a + b;
}
*/
import _ "unsafe"
import "fmt"

func Foo() {
        fmt.Println(( /*line :13:14*/_Cfunc_floor /*line :13:20*/)(( /*line :13:22*/_Cfunc_add /*line :13:26*/)(1, 2.1)))
}

---------
adjusted position: /home/ldez/.cache/go-build/52/52709e04346f162eeabb16dbe9df656d978effaafca09c371368789c95dd32d3-d non adjusted position: /home/ldez/.cache/go-build/52/52709e04346f162eeabb16dbe9df656d978effaafca09c371368789c95dd32d3-d

adjusted position file: /home/ldez/.cache/go-build/52/52709e04346f162eeabb16dbe9df656d978effaafca09c371368789c95dd32d3-d 
 package cgo
//go:cgo_import_dynamic __libc_start_main __libc_start_main#GLIBC_2.34 "libc.so.6"
//go:cgo_import_dynamic _ITM_deregisterTMCloneTable _ITM_deregisterTMCloneTable ""
//go:cgo_import_dynamic __gmon_start__ __gmon_start__ ""
//go:cgo_import_dynamic _ITM_registerTMCloneTable _ITM_registerTMCloneTable ""
//go:cgo_import_dynamic __cxa_finalize __cxa_finalize#GLIBC_2.2.5 "libc.so.6"
//go:cgo_import_dynamic _ _ "libc.so.6"

non adjusted position file: /home/ldez/.cache/go-build/52/52709e04346f162eeabb16dbe9df656d978effaafca09c371368789c95dd32d3-d 
 package cgo
//go:cgo_import_dynamic __libc_start_main __libc_start_main#GLIBC_2.34 "libc.so.6"
//go:cgo_import_dynamic _ITM_deregisterTMCloneTable _ITM_deregisterTMCloneTable ""
//go:cgo_import_dynamic __gmon_start__ __gmon_start__ ""
//go:cgo_import_dynamic _ITM_registerTMCloneTable _ITM_registerTMCloneTable ""
//go:cgo_import_dynamic __cxa_finalize __cxa_finalize#GLIBC_2.2.5 "libc.so.6"
//go:cgo_import_dynamic _ _ "libc.so.6"

---------
```

</details>

**Pull request merged**